File: evaluation/QuixBugs/java_testcases/DETECT_CYCLE_TEST.java
package java_testcases;
import java.util.*;

import java_programs.DETECT_CYCLE;
import java_programs.Node;


/**
 * Driver for testing DETECT_CYCLE class
 */
public class DETECT_CYCLE_TEST {
    public static void main(String[] args) throws Exception {
        // Case 1: 5-node list input with no cycle
        // Expected Output: Cycle not detected!
        Node node1 = new Node("1");
        Node node2 = new Node("2", node1);
        Node node3 = new Node("3", node2);
        Node node4 = new Node("4", node3);
        Node node5 = new Node("5", node4);

        if (DETECT_CYCLE.detect_cycle(node5)) {
            System.out.println("Cycle detected!");
        } else {
            System.out.println("Cycle not detected!");
        }

        // Case 2: 5-node list input with cycle
        // Expected Output: Cycle detected!
        node1.setSuccessor(node5);

        if (DETECT_CYCLE.detect_cycle(node5)) {
            System.out.println("Cycle detected!");
        } else {
            System.out.println("Cycle not detected!");
        }

        // Case 3: 2-node list with cycle
        //  Expected Output: Cycle detected!
        node1.setSuccessor(node2);
        if (DETECT_CYCLE.detect_cycle(node2)) {
            System.out.println("Cycle detected!");
        } else {
            System.out.println("Cycle not detected!");
        }

        // Case 4: 2-node list with no cycle
        //  Expected Output: Cycle not detected!
        Node node6 = new Node("6");
        Node node7 = new Node("7", node6);
        if (DETECT_CYCLE.detect_cycle(node7)) {
            System.out.println("Cycle detected!");
        } else {
            System.out.println("Cycle not detected!");
        }

        // Case 5: 1-node list
        // Expected Output: Cycle not detected!
        Node node = new Node("0");
        if (DETECT_CYCLE.detect_cycle(node)) {
            System.out.println("Cycle detected!");
        } else {
            System.out.println("Cycle not detected!");
        }
    }
}

File: evaluation/other/QuixBugs/correct_python_programs/detect_cycle_test.py
from .node import Node
from .detect_cycle import detect_cycle


"""
Driver to test reverse linked list
"""
def main():
    # Case 1: 5-node list input with no cycle
    # Expected Output: Cycle not detected!
    node1 = Node(1)
    node2 = Node(2, node1)
    node3 = Node(3, node2)
    node4 = Node(4, node3)
    node5 = Node(5, node4)

    if detect_cycle(node5):
        print("Cycle detected!", end=" ")
    else:
        print("Cycle not detected!", end=" ")
    print()

    # Case 2: 5-node list input with cycle
    # Expected Output: Cycle detected!
    node1.successor = node5

    if detect_cycle(node5):
        print("Cycle detected!", end=" ")
    else:
        print("Cycle not detected!", end=" ")
    print()

    # Case 3: 2-node list with cycle
    # Expected Output: Cycle detected!
    node1.successor = node2

    if detect_cycle(node2):
        print("Cycle detected!", end=" ")
    else:
        print("Cycle not detected!", end=" ")
    print()

    # Case 4: 2-node list with no cycle
    # Expected Output: Cycle not detected!
    node6 = Node(6)
    node7 = Node(7, node6)

    if detect_cycle(node7):
        print("Cycle detected!", end=" ")
    else:
        print("Cycle not detected!", end=" ")
    print()

    # Case 5: 1-node list
    # Expected Output: Cycle not detected!
    node = Node(0)
    if detect_cycle(node):
        print("Cycle detected!", end=" ")
    else:
        print("Cycle not detected!", end=" ")
    print()

if __name__ == "__main__":
    main()

